apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/CommonUtils.cs

```diff
@@ -43,31 +44,17 @@ internal static class CommonUtils
 
         /// <summary>
         /// Determines the correct project id from a string project id and a <see cref="MonitoredResource"/>.
-        /// <list type="bullet">
-        /// <term>
-        /// If the string project id is not null and the MonitoredResource does not contain a project id
-        /// then the string project id is retruned.
-        /// </term>
-        /// <term>
-        /// If the string project id is not null and the MonitoredResource contains a project id
-        /// then the string project id is retruned.  We do this as the explictly set project id should be
-        /// defaulted too.
-        /// </term>
-        /// <term>
-        /// If the string project id is null and the MonitoredResource does contains a project id
-        /// then the project id from the MonitoredResource is retruned.
-        /// </term>
-        /// <term>
-        /// If the string project id is null and the MonitoredResource does not contain a project id
-        /// then an <see cref="InvalidOperationException"/> is thrown.
-        /// </term>
-        /// </list>
+        /// If the specified project id is not null, it is returned. Otherwise, if the project id of
+        /// the MonitoredResource is not null, it is returned. It both are null,
+        /// an <see cref="InvalidOperationException"/> is thrown.
         /// </summary>
         /// <param name="projectId">The Google Cloud project ID.  Can be null.</param>
-        /// <param name="monitoredResource">The monitored resource.</param>
+        /// <param name="monitoredResource">Optional, The monitored resource. If unset the monitored resource will
+        ///     be auto detected.</param>
         /// <returns>The Google Cloud project ID.</returns>
-        internal static string GetAndCheckProjectId(string projectId, MonitoredResource monitoredResource)
+        internal static string GetAndCheckProjectId(string projectId, MonitoredResource monitoredResource = null)
         {
+            monitoredResource = monitoredResource ?? MonitoredResourceBuilder.FromPlatform();
```

Comment:
Actually, my idea was slightly different here. I was thinking that if the `projectId` was specified in this method but `monitoredResource` was not, you wouldn't have to bother getting it. You could just return the `projectId` immediately. Then in all the call sites, you wouldn't need `projectId ??`. The downside is that you can't do the debug check below to see if the project id matches, but you have that situation now anyway as it is.